### ISSUE TITLE
Unable to run pipeline using the Python API Client (KFP)

### ISSUE THREAD

**Issue by sahilmalav** [kubeflow/pipelines#2430]
Uploaded a pipeline successfully using the python api client. Got the following response:
_{'created_at': datetime.datetime(2019, 10, 18, 6, 55, 9, tzinfo=tzutc()),
 'description': None,
 'error': None,
 'id': '45e153d7-8826-4738-bcaa-71556d2f95f9',
 'name': 'testing',
 'parameters': [{'name': 'repo-name', 'value': None},
                {'name': 'branch-name', 'value': None},
                {'name': 'commit', 'value': None},
                {'name': 'output-path', 'value': None},
                {'name': 'input-path', 'value': None}],
 'url': None}_

Then created an experiment using the api client, which too was successful.
Next, I want to run my pipeline using the client. For this, I wrote the following python code:

**import kfp
cl = kfp.Client(host='http://172.16.6.51:31380/pipeline')

runname = 'apitesting'
params = {
                "repo-name": "irrs_demo_repo",
                "branch-name": "experiment",
                "commit": "c0c813f1d80643799fe8eac9a57f4af5",
                "output-path": "/data",
                "input-path": "/dataset/dataxlsx"
}
pip_id = '45e153d7-8826-4738-bcaa-71556d2f95f9'
exp_id = '873fef4a-ddb2-4f74-811f-0f92b0d0464e' 
r = cl.run_pipeline(experiment_id=exp_id, job_name=runname, pipeline_id=pip_id, params=params)
print(r)**

Response was : 
**Traceback (most recent call last):
  File "/opt/xpresso.ai/xpresso/ai/server/controller/project_management/kubernetes/test.py", line 123, in <module>
    r = cl.run_pipeline(experiment_id=eid, job_name=runname, pipeline_id=pip_id, params=params)
  File "/usr/local/lib/python3.7/dist-packages/kfp/_client.py", line 313, in run_pipeline
    response = self._run_api.create_run(body=run_body)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/api/run_service_api.py", line 151, in create_run
    (data) = self.create_run_with_http_info(body, **kwargs)  # noqa: E501
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/api/run_service_api.py", line 228, in create_run_with_http_info
    collection_formats=collection_formats)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/api_client.py", line 330, in call_api
    _preload_content, _request_timeout)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/api_client.py", line 161, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/api_client.py", line 373, in request
    body=body)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/rest.py", line 275, in POST
    body=body)
  File "/usr/local/lib/python3.7/dist-packages/kfp_server_api/rest.py", line 228, in request
    raise ApiException(http_resp=r)
kfp_server_api.rest.ApiException: (400)
Reason: Bad Request
HTTP response headers: HTTPHeaderDict({'x-powered-by': 'Express', 'content-type': 'application/json', 'trailer': 'Grpc-Trailer-Content-Type', 'date': 'Fri, 18 Oct 2019 07:07:52 GMT', 'x-envoy-upstream-service-time': '27', 'server': 'istio-envoy', 'transfer-encoding': 'chunked'})
HTTP response body: {"error":"Failed to create a new run.: Failed to convert run model: Unable to convert resource references.: Failed to convert relationship: Invalid input error: Unsupported resource relationship: UNKNOWN_RELATIONSHIP","message":"Failed to create a new run.: Failed to convert run model: Unable to convert resource references.: Failed to convert relationship: Invalid input error: Unsupported resource relationship: UNKNOWN_RELATIONSHIP","code":3,"details":[{"@type":"type.googleapis.com/api.Error","error_message":"Unsupported resource relationship: UNKNOWN_RELATIONSHIP","error_details":"Failed to create a new run.: Failed to convert run model: Unable to convert resource references.: Failed to convert relationship: Invalid input error: Unsupported resource relationship: UNKNOWN_RELATIONSHIP"}]}**

I tried a lot of different configurations, but I keep getting this error. 

Versions:
Kubeflow 0.6.1 
kfp 0.1.31.2 
kfp_server_api 0.1.26

**Comment by Ark-kun** [comment]
You might want to update the Kubeflow backend version. 
@IronPan I think this is the case of that fixed bug with SQL truncations.

**Comment by sahilmalav** [comment]
I debugged the code. In the client's run_pipeline method, the reference object generated has the 'relationship' value as None. It should be 'OWNER' instead.

**Comment by Ark-kun** [comment]
AFAIK, `relationship` is optional.

**Comment by sahilmalav** [comment]
Doesn't seem so. My issue was resolved when I manually set it to 'OWNER'.

**Comment by Ark-kun** [comment]
> the reference object generated has the 'relationship' value as None.

AFAIK it's set to `OWNER` here:

https://github.com/kubeflow/pipelines/blob/d1c32e592dd9d6beb216863e6ed2d4ab24d43573/sdk/python/kfp/_client.py#L310

**Comment by Ark-kun** [comment]
Does everything work when you call `run_pipeline` on a pipeline package instead of a pipeline id?

**Comment by IronPan** [comment]
@jingzhang36 could you take a look this?

**Comment by sahilmalav** [comment]
> Does everything work when you call `run_pipeline` on a pipeline package instead of a pipeline id?

@Ark-kun : No. Same issue irrespective of how I make the call. The only fix that worked for me was manually setting the relationship to OWNER.

**Comment by Ark-kun** [comment]
> The only fix that worked for me was manually setting the relationship to OWNER.

Can you explain what exact change you had to do?

**Comment by jingzhang36** [comment]
Here when using swagger-codegen-cli 2.4.7 and 2.4.1 to generate frontend apis. Here is an example interface:

listRuns(page_token?: string, page_size?: number, sort_by?: string, resource_reference_key_type?: 'UNKNOWN_RESOURCE_TYPE' | 'EXPERIMENT' | 'JOB' | 'PIPELINE', resource_reference_key_id?: string, filter?: string, options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<ApiListRunsResponse>

Please note the input parameter resource_reference_key_type; its type is 'UNKNOWN_RESOURCE_TYPE' | 'EXPERIMENT' | 'JOB' | 'PIPELINE'. However, in contrast, our master repo has this same method's interface as 

listRuns(page_token?: string, page_size?: number, sort_by?: string, resource_reference_key_type?: string, resource_reference_key_id?: string, filter?: string, options: any = {}): FetchArgs
https://github.com/kubeflow/pipelines/blob/5043ec78afc63506ce70fca01d2936928cb0fedb/frontend/src/apis/run/api.ts#L710

And apparently, the type for resource_reference_key_type is simply string instead of 'UNKNOWN_RESOURCE_TYPE' | 'EXPERIMENT' | 'JOB' | 'PIPELINE'.

The caller to this method using "ApiResourceType.EXPERIMENT.toString()" https://github.com/kubeflow/pipelines/blob/2d6a314e549cf27dd260c9777c4da7314caa9a80/frontend/src/pages/ExperimentList.tsx#L166 works with the old interface but not the new one generated by 2.4.7 or 2.4.1.

In short, our master repo's generated code needs to be updated since we move to newer version of swagger-codegen-cli

**Comment by jingzhang36** [comment]
BTW, the master repo's generated code seems from swagger-codegen-cli.jar 2.3.1

**Comment by jingzhang36** [comment]
And for the exact error in this message, I feel we also need to update _client.py in sdk to use string 'OWNER' and 'EXPERIMENT' string as well, similarly to what updates we do in frontend.

**Comment by jingzhang36** [comment]
@Ark-kun can check if the sdk change here https://github.com/kubeflow/pipelines/pull/2453/files#diff-1f65a44a2187458da0f00e0ebbe730e5 is ok or not?

**Comment by Ark-kun** [comment]
kfp-server-api v0.1.18.3 was supposed to be generated using swagger v2.4.7.
I've checked it and it works.

kfp-server-api v0.1.26 does not work.

kfp-server-api v0.1.18.3 is the version specified by KFP in setup.py.

**Comment by jingzhang36** [comment]
#Hacktoberfest

**Comment by Ark-kun** [comment]
> kfp_server_api 0.1.26

This version is outside of the KFP supported version range (from `setup.py`): `'kfp-server-api >= 0.1.18, <= 0.1.25'`

Please downgrade you kfp_server_api by running `pip3 install kfp_server_api==0.1.18.3 --upgrade`.

It might be that the person creating `0.1.26` used a wrong script or swagger version. I'll investigate.

**Comment by Ark-kun** [comment]
@sahilmalav Can you please verify that kfp_server_api==0.1.18.3 works for you?

**Comment by sahilmalav** [comment]
I'm currently out on a vacation and don't have my laptop with me. I'll only be able to check next week.

**Comment by Ark-kun** [comment]
> I'm currently out on a vacation and don't have my laptop with me. I'll only be able to check next week.

Good luck on your vacation. I've verified that 0.1.18.3 works while 0.1.26 does not.
